### PR TITLE
Add clouseau search index task

### PR DESCRIPTION
**Pull request by javierg**
This task will allow to add searches with Clouseau Engine

https://docs.couchdb.org/en/3.2.0/ddocs/search.html#index-functions

The create index task works by reading the folder:

`db/searches`

And parsing each JS file it encounters in the folder into a single `_design/base_filename` document

The only supported extension is `.js`

And the index will be named the same as the filename, so `db/searches/my_index.js`
will be queried like: `_design/_my_index/_search/my_index?q=*:*`

### CODE REVIEW

File: lib/tasks/db.rake [hunk 1]
```diff
@@ -61,6 +61,77 @@ namespace :db do
...
+        if data[:indexes].to_json == doc[:indexes].to_json
+          puts "\e[32mSearch index #{name} is up to date.\e[0m" unless silent
+          {}
+        else
+          puts "\e[36mSearch Index #{name} will be updated.\e[0m" unless silent
+          doc.merge!(data)
+        end
+      end
+
+      res = Dolly::Document.connection.request :post, "#{db}_bulk_docs", docs: docs
```
Comment (alejandrodevs) [inline lib/tasks/db.rake:116]: I think you missed a `/`. If a db name is passed the url should be `#{db}/_bulk_docs`? Or am I missing something?

File: lib/dolly/search_index.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      ROWS_KEY = :rows
+      DESIGN_PREFIX = '_design/'
+
+      def_delegators :connection, :get, :post
+
+      def all
+        get(DESIGN)[:indexes]
+      end
+
+      def create(name, fields)
```
Comment (alejandrodevs) [inline lib/dolly/search_index.rb]: IMO these actions should always receive a db name. Do we want to create global search indexes?

File: lib/tasks/db.rake (hunk 1 above)
Comment (javierg) [reply lib/tasks/db.rake:116]: Nop, it is like this so if no db is setup dolly will peak the default db and no initial `/` is required

File: lib/dolly/search_index.rb (hunk 2 above)
Comment (javierg) [reply lib/dolly/search_index.rb]: ah! this file is no used actually... let me removed

File: lib/tasks/db.rake (hunk 1 above)
Comment (alejandrodevs) [reply lib/tasks/db.rake:116]: Nevermind, my bad.

**Review by alejandrodevs — APPROVED** [review]
🚢

File: lib/tasks/db.rake (hunk 1 above)
Comment (alejandrodevs) [reply lib/tasks/db.rake:116]: I didnt see you add the / in the db variable.

**Pull request merged**